Source/NetworkActivityIndicatorManager.swift

```diff
@@ -87,6 +87,8 @@ public class NetworkActivityIndicatorManager {
 
     private var activityIndicatorState: ActivityIndicatorState = .notActive {
         didSet {
+            lock.lock() ; defer { lock.unlock() }
```

Comment:
Is there a path to the `didSet` that isn't through a method that's already locked? They're run synchronously, so they'll be run before the `defer` unlocks I think.

Reply:
Good catch. I couldn't remember if `didSet` was run synchronously or not. There's no path that isn't currently locked, so this change technically isn't necessary then. The bigger question is whether we'd want to protect against a future change that mutated the state that wasn't locked. I could go either way on that one since I updated the lock to be recursive, but I think I'll remove it for now since it's not technically needed.

Reply:
Dropped this unnecessary lock in 0e657077.